west: runners: jlink: prefer .hex over .bin

JordanYates opened the pull request:
Update the jlink runner to prefer flashing .hex files instead of .bin.
This can increase programming speed in cases where there are large
amounts of padding in an application.

This PR is the alternative to adding a new `--hex-file` option discussed in #31304

Signed-off-by: Jordan Yates <jordan.yates@data61.csiro.au>

mbolivar-nordic reviewed (changes requested):
Thanks for this. I think this PR would be good if we were just looking to flash .hex only, but not everybody enables CONFIG_BUILD_OUTPUT_HEX. Can you please modify it so it prefers the .hex when it is available, but still falls back on the .bin? Basically doing the same thing as the pyocd runner here in the jlink runner.

Commit:
west: runners: jlink: prefer .hex over .bin

Update the jlink runner to prefer flashing .hex files instead of .bin.
This can increase programming speed in cases where there are large
amounts of padding in an application.

Signed-off-by: Jordan Yates <jordan.yates@data61.csiro.au>

mbolivar-nordic reviewed (approved):
Thanks, nice improvement. Tested with and without CONFIG_BUILD_OUTPUT_HEX=y.
